runtime/lua/vim/diagnostic.lua

```diff
@@ -372,14 +372,15 @@ local function schedule_display(namespace, bufnr, args)
   bufs_waiting_to_update[bufnr][namespace] = args
 
   local key = make_augroup_key(namespace, bufnr)
-  local group = vim.api.nvim_create_augroup(key, { clear = true })
   if not registered_autocmds[key] then
+    local group = vim.api.nvim_create_augroup(key, { clear = true })
     vim.api.nvim_create_autocmd(insert_leave_auto_cmds, {
       group = group,
       buffer = bufnr,
       callback = function()
         execute_scheduled_display(namespace, bufnr)
       end,
+      desc = 'Diagnostic ' .. table.concat(insert_leave_auto_cmds, ','),
```

Comment:
The information about what events the autocommand is attached is redundant. Maybe the description could be something like `vim.diagnostic: display diagnostics`?

Reply:
done :)